scripts/vehicle_node.py

```diff
@@ -0,0 +1,301 @@
+        rospy.Timer(rospy.Duration(0.05), self.update_position)
+
+        #Subscriber for to teleop key
+        rospy.Subscriber("/turtlebot_teleop/cmd_vel", Twist, self.velocity_cmd_callback)
+
+        #Publisher for PoseStamped() Message
+        self.pose_publisher = rospy.Publisher("/move_vehicle/cmd", PoseStamped, queue_size=10)
+
+        #Publisher for a Float32MultiArray() Message
+        self.pose_array_publisher = rospy.Publisher("/move_vehicle/cmd_array", Float32MultiArray, queue_size=10)
+
+        #Publisher for PointCloud2 terrain map
+        self.point_cloud_publisher = rospy.Publisher("/point_cloud", PointCloud2, latch = True, queue_size=10)
+
+        #Publisher for Vehicle Marker
+        self.grid_publisher = rospy.Publisher('/grid_marker', Marker, queue_size=10 )
+
+        #Keeps program running until interrupted
+        rospy.spin()
+
+
+    def create_terrain_map(self):
+
+        """
+        Creates a list of points from an array of gaussian noise given a specified height and width 
+        """
+
+        #Get parameters
+        mu = rospy.get_param('/mu')
+        sigma = rospy.get_param('/sigma')
+        sigma_filt = rospy.get_param('/sigma_filt')
+        height = rospy.get_param('/height')
+        width = rospy.get_param('/width')
+        resolution =  rospy.get_param('/resolution')
+
+        #Fill array of size h x w with Gaussian Noise given the height and width of the point cloud
+        h = int(((height*2) + 1)*(resolution*1000))
+        w = int(((width*2) + 1)*(resolution*1000))
+        grid_map = (h,w)
+        gaussian_array = np.random.normal(mu, sigma, grid_map)
+
+        #Filter the array to smoothen the variation of the noise
+        gaussian_array = gaussian_filter(gaussian_array, sigma_filt)
+
+        #Create an list and fill the list with x, y values and z the values will be the filtered gaussian noise
+        # TODO perform these calculations without for loops
+        incr_x = 0
+        incr_y = 0
+        cloud = []
+        for x in np.arange(-height, (height+1), resolution):
+            for y in np.arange(-width, (width+1), resolution):
+                innerlist = []
+                innerlist.append(x)
+                innerlist.append(y)
+                innerlist.append(gaussian_array[incr_x][incr_y])
+                cloud.append(innerlist)
+                incr_y+=1
+            incr_x+= 1
+            incr_y = 0
+
+        return cloud
+
+
+    def velocity_cmd_callback(self, data):
+        
+        """
+        Updates the most recent command velocity to a twist variable
+        """
+        self.lock.acquire()
+        try:
+            self.twist = data
+        finally:
+            self.lock.release()
+
+
+    def update_position(self, event):
+
+        """
+        Computes the pose of the vehicle within a given footprint height and width 
+        """
+
+        #Load parameters
+        resolution =  rospy.get_param('/resolution')
```

Comment:
If these parameters aren't being changed during runtime of the node we should load it as member variables rather than fetching from the param server continuously